.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,120 @@
+name: "Cached build"
+description: "Setup the build using cache"
+inputs:
+  ghc:
+    description: "Ghc version"
+    required: true
+  cabal:
+    description: "Cabal version"
+    required: false
+    default: "3.6"
+  os:
+    description: "Operating system: Linux, Windows or macOS"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - uses: haskell/actions/setup@v1
+      id: HaskEnvSetup
+      with:
+        ghc-version  : ${{ inputs.ghc   }}
+        cabal-version: ${{ inputs.cabal }}
+        enable-stack: false
+
+    - if: inputs.os == 'Windows'
+      name: (Windows) Platform config
+      run: |
+        echo "CABAL_PKGS_DIR=C:\\cabal\\packages" >> $GITHUB_ENV
+      shell: bash
+    - if: ( inputs.os == 'Linux' ) || ( inputs.os == 'macOS' )
+      name: (Linux,macOS) Platform config
+      run: |
+        echo "CABAL_PKGS_DIR=~/.cabal/packages" >> $GITHUB_ENV
+      shell: bash
+
+    # Needs to be before Cache Cabal so the cache can detect changes to the modified cabal.project file
+    - if: inputs.ghc == '9.0.1' || inputs.ghc == '9.2.1'
+      name: (GHC 9.0/9.2) Use modified `cabal.project`
+      env:
+        GHCVER: ${{ inputs.ghc }}
+      run: |
+        # File has some protections preventing regular `rm`.
+        # (most probably sticky bit is set on $HOME)
+        # `&&` insures `rm -f` return is positive.
+        # Many platforms aslo have `alias cp='cp -i'`.
+        rm -f -v cabal.project && cp -v cabal-ghc${GHCVER//./}.project cabal.project
```

Comment:
Checking escaping in-build:
https://github.com/haskell/haskell-language-server/runs/4707587679?check_suite_focus=true#step:3:60:
```
removed 'cabal.project'
'cabal-ghc901.project' -> 'cabal.project'
```

Great, works as intended.